wgsl/index.bs

```diff
@@ -2036,16 +2036,27 @@ Some consequences of the restrictions structure member and array element types a
     | [=syntax/attribute=] * [=syntax/member_ident=] [=syntax/colon=] [=syntax/type_decl=]
 </div>
 
-WGSL defines the following attributes that can be applied to structure members:
+The following attributes can be applied to structure members:
  * [=attribute/align=]
  * [=attribute/builtin=]
  * [=attribute/location=]
  * [=attribute/interpolate=]
  * [=attribute/invariant=]
  * [=attribute/size=]
 
-Note: Layout attributes may be required if the structure type is used
-to define a [=uniform buffer=] or a [=storage buffer=]. See [[#memory-layouts]].
+Attributes [=attribute/builtin=] [=attribute/location=] [=attribute/interpolate=], and [=attribute/invariant=]
+are [=IO attributes=].
+An [=IO attribute=] on a member of a structure |S| has effect only when
+|S| is used as the type of a [=formal parameter=] or [=return type=] of an [=entry point=].
+See [[#pipeline-inputs-outputs]].
+
+Attributes [=attribute/align=] and [=attribute/size=] are [=layout attributes=],
+and may be required if the structure type is used to
+define a [=uniform buffer=] or a [=storage buffer=].
+See [[#memory-layouts]].
+
+When another type |T| is defined as a [=type alias=] for structure type |S|,
```

Comment:
I'd suggest putting this in the type alias section and generalizing that any attributes on the type or members of the composite are captured by the alias.

Reply:
An attribute can't be applied to a type itself. (The last case was "block", and that's gone).